src/components/Pages/Search/index.js

```diff
@@ -14,33 +16,150 @@ import SearchResult from 'components/Pages/Search/searchResult.js'
 
 const SearchPage = () => {
 
-  const { searchIndex } = useRouteData();
+  const intl = useIntl();
+  let { searchIndex } = useRouteData();
+  let searchedTerm = ""
 
-  const title = "Search" // TODDO: ⚠️useIntl
+  useEffect(() => {
+    const input = document.getElementById("coa-search_input")
+    input.focus()
+    const filteredSearch = searchWorker(searchIndexWithUrl, input.value)
+    setSearchResults(filteredSearch)
+    if (typeof window !== 'undefined') {
+      window.location.hash = input.value.toLocaleLowerCase()
+    }
+  }, [searchIndex]);
+
+  // Check the url for a search term to apply
+  if (typeof window !== 'undefined' && window.location.hash.length > 1) {
+    // TODO: ...when filters are added to the url, create an array instead and use the '&' standard.
+    // const queryArr = decodeURIComponent(window.location.hash.split("#")[1]).split("&")
+    // searchedTerm = queryArr[0]
+    searchedTerm = decodeURIComponent(window.location.hash.split("#")[1])
+  }
+
+  // hook makes our input dynamic (useful for "as-you-type" filtering)
+  let [ searchString, setSearchString ] = useState(searchedTerm)
+
+  // Don't show pages without Url (there seems to be some pages that are 'live', but without a url - catch those here...)
+  const searchIndexWithUrl = searchIndex.filter( page => page.janisUrls.length > 0)
+
+  let [ searchResults, setSearchResults ] = useState(searchIndexWithUrl)
+
+  const searchButtonPressed = function() {
+    loader.start({
+      contentId: 'coa-search_results',
+      loaderId: 'coa-search_loading_wheel',
+    })
+    setTimeout(function(){
+      if (typeof window !== 'undefined') {
+        window.location.hash = searchString.toLocaleLowerCase()
+      }
+      const filteredSearch = searchWorker(searchIndexWithUrl, searchString)
+      setSearchResults(filteredSearch)
+      loader.end()
+    },1000)
+  }
```

Comment:
Ok, ummm... hear me out. Yes, we're faking a delay here. One reason, as we've already done with pagination, is that we want to "signal" the user that a change has taken place. The other (as to the full second) is that potentially (and eventually as more pages are added) there may be a _slight_ delay for more processing. Or, as we add an api, there will definitely be a request delay. Call it a "pre-emptive regression buffer" ™️. Or a "delay reserve"™️.

And, of course, open to changing.

Reply:
Let's take out the delay. Unless this is specifically mandated by design, then we can push back. Instantaneous search results are not a problem. If Google added a second of delay to every search result, we'd all be developing on Bing Chrome.